chia/wallet/util/new_peak_queue.py

```diff
@@ -53,18 +53,28 @@ def __ge__(self, other):
 class NewPeakQueue:
     def __init__(self, inner_queue: asyncio.PriorityQueue):
         self._inner_queue: asyncio.PriorityQueue = inner_queue
+        self._pending_data_process_items: int = 0
 
     async def subscribe_to_coin_ids(self, coin_ids: List[bytes32]):
+        self._pending_data_process_items += 1
         await self._inner_queue.put(NewPeakItem(NewPeakQueueTypes.COIN_ID_SUBSCRIPTION, coin_ids))
 
     async def subscribe_to_puzzle_hashes(self, puzzle_hashes: List[bytes32]):
+        self._pending_data_process_items += 1
         await self._inner_queue.put(NewPeakItem(NewPeakQueueTypes.PUZZLE_HASH_SUBSCRIPTION, puzzle_hashes))
 
     async def full_node_state_updated(self, coin_state_update: CoinStateUpdate, peer: WSChiaConnection):
+        self._pending_data_process_items += 1
         await self._inner_queue.put(NewPeakItem(NewPeakQueueTypes.FULL_NODE_STATE_UPDATED, (coin_state_update, peer)))
 
     async def new_peak_wallet(self, new_peak: NewPeakWallet, peer: WSChiaConnection):
         await self._inner_queue.put(NewPeakItem(NewPeakQueueTypes.NEW_PEAK_WALLET, (new_peak, peer)))
 
     async def get(self) -> NewPeakItem:
-        return await self._inner_queue.get()
+        item: NewPeakItem = await self._inner_queue.get()
+        if item.item_type != NewPeakQueueTypes.NEW_PEAK_WALLET:
+            self._pending_data_process_items -= 1
+        return item
+
+    def has_pending_data_process_items(self) -> bool:
```

Comment:
is it possible to never receive an update to one of those subscriptions? like maybe a peer is overwhelmed etc? how would the wallet recover from something like that since counter would never get to zero?

Reply:
this is a local counter for local queue items. We pull an item from the queue (decreasing the counter), and then we ask the full node. If the full node does not respond, that's it, we forget about it. The item was still pulled from the queue. Does that make sense?

Reply:
ah, yes 🤦‍♂️